File: 0020-valid-parentheses-有效的括号/0020.cpp
class Solution {
public:
    bool isValid(string s) {
        //unsigned small = 0 ,middle = 0, large=0;   //small,middle,large
        vector<int> v ={};
        for(auto a1: s){
            switch(a1){
                case '(':
                    v.push_back(0);
                    break;
                case '[':
                    v.push_back(1);
                    break;
                case '{':
                    v.push_back(2);
                    break;
                case ')':
                    if(!v.empty() && v.back()==0){
                        v.pop_back();
                        break;
                    }else{
                        return false;
                    }
                case ']':
                    if(!v.empty() && v.back()==1){
                        v.pop_back();
                        break;
                    }else{
                        return false;
                    }
                case '}':
                    if(!v.empty() && v.back()==2){
                        v.pop_back();
                        break;
                    }else{
                        return false;
                    }
                default:
                    break;
            }
        }
        if(!v.empty()){return false;}
        return true;
    }
};

File: 0020-valid-parentheses-有效的括号/0020.py
class Solution:
    def isValid(self, s: str) -> bool:
        t = ['default']
        for c in s:
            if c == '(':
                t.append(')')
            elif c == '{':
                t.append('}')
            elif c == '[':
                t.append(']')
            elif t[-1] == c:
                t.pop()
            else:
                return False
        if len(t) == 1:
            return True
        return False